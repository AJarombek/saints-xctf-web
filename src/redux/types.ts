/**
 * Types for the Redux store.
 * @author Andrew Jarombek
 * @since 7/23/2020
 */

export type RootState = {
    auth: AuthState,
    logs: LogsState
}

export type LogsState = {
    isFetching: boolean;
    didInvalidate: boolean;
    lastUpdated: number;
    items: Logs;
    feeds: LogFeeds;
    newComments: NewComments;
}

export type AuthState = {
    auth: Auth;
    user: Users;
}

export interface Meta {
    isFetching?: boolean;
    lastUpdated?: number;
    didInvalidate?: boolean;
    serverError?: string;
}

export type Auth = {
    isFetching?: boolean;
    lastUpdated?: number;
    signedInUser?: string;
    status?: string;
}

export type Users = {
    [key: string]: UserMeta;
}

export interface UserMeta extends User, Meta {}

export interface User {
    activation_code?: string;
    class_year?: number;
    deleted?: string;
    description?: string;
    email?: string;
    favorite_event?: string;
    first?: string;
    last?: string;
    last_signin?: string;
    location?: string;
    member_since?: string;
    password?: string;
    salt?: string;
    subscribed?: string;
    username?: string;
    week_start?: string;
}

enum LogType { RUN = "run", BIKE = "bike", SWIM = "swim", OTHER = "other" }
enum Metric { MILES = "miles", KILOMETERS = "kilometers", METERS = "meters" }

export type Log = {
    log_id: number;
    username: string;
    first: string;
    last: string;
    name: string;
    date: string;
    type: LogType;
    distance?: number;
    metric?: Metric;
    miles?: number;
    time?: string;
    pace?: string;
    feel: number;
    location?: string;
    description?: string;
    timeCreated?: string;
    comments?: Comment[];
}

export type Logs = {
    [key: string]: Record<string, Log>;
}

export type LogFeeds = {
    [key: string]: {
        filterBy: string;
        bucket: string;
        pages: LogFeedPages
    };
}

export type LogFeedPages = {
    [key: string]: LogFeedPage
}

export type LogFeedPage = {
    isFetching: boolean;
    lastUpdated: number;
    items: Log[];
    serverError: string;
}

export type Comment = {
    comment_id: number;
    username: string;
    first: string;
    last: string;
    log_id: number;
    time: string;
    content: string;
}

export type NewComments = {
    [key: string]: {
        isFetching: boolean;
        lastUpdated: number;
        created?: boolean;
        serverError?: string;
    }
}
