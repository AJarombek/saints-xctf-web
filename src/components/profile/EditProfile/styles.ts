/**
 * JSS styles for the EditProfile component.
 * @author Andrew Jarombek
 * @since 10/18/2020
 */

import color from 'color';
import Mixins from '../../../styles/mixins';

export default {
  editProfile: {},
  title: {
    ...Mixins.formTitle(),
    fontSize: 20
  },
  form: {
    backgroundColor: color('#E3E3E3').alpha(0.8).string(),
    width: '100%',
    maxWidth: '700px',
    margin: '25px auto',
    borderRadius: '6px',
    padding: '20px'
  },
  inputError: {
    ...Mixins.inputError()
  },
  inputTitle: {
    ...Mixins.inputTitle()
  },
  twoInputs: {
    display: 'flex'
  },
  textArea: {
    border: 'none',
    width: '100%',
    padding: '6px 20px'
  },
  radioGroup: {
    display: 'flex'
  },
  radio: {
    margin: '0 20px'
  },
  firstNameInput: {
    ...Mixins.rightInput(50)
  },
  lastNameInput: {
    ...Mixins.leftInput(50)
  },
  emailInput: {
    ...Mixins.rightInput(80)
  },
  classYearInput: {
    ...Mixins.leftInput(20),

    '& input': {
      width: 'auto'
    }
  },
  locationInput: {
    ...Mixins.rightInput(50)
  },
  favoriteEventInput: {
    ...Mixins.leftInput(50)
  },
  actions: {
    ...Mixins.formActions()
  },
  submitButton: {
    marginRight: 20
  },
  cancelButton: {
    marginRight: 10
  },
  profilePictureContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  picture: {
    ...Mixins.profilePicture(),
    marginLeft: 0
  },
  '@media screen and (max-width: 720px)': {
    title: {
      width: 'auto'
    }
  }
};
