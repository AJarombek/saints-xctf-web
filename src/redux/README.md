### Overview

Configure Redux as the state management system for the application.

### Files

| Filename                 | Description                                                                |
|--------------------------|----------------------------------------------------------------------------|
| `modules`                | Redux modules containing Reducers, Actions, and Action Creators.           |
| `appState.js`            | Sample application state maintained by Redux.                              |
| `store.js`               | Configuration and creation of the Redux store.                             |
| `types.ts`               | Typescript types used throughout the application.                          |

### Resources

1) [React/Redux Example App #1](https://github.com/erikras/react-redux-universal-hot-example/tree/master/src/redux)
2) [React/Redux Example App #2](https://github.com/goopscoop/ga-react-tutorial/tree/6-reduxActionsAndReducers/src/redux)
3) [React/Redux Example App #3](https://github.com/gothinkster/react-redux-realworld-example-app/tree/master/src)
4) [Redux Ducks Spec](https://github.com/erikras/ducks-modular-redux)
5) [React Router/Redux Setup Steps](https://github.com/supasate/connected-react-router#usage)
6) [Redux Hot Reloading](https://redux.js.org/recipes/configuring-your-store/#hot-reloading)
7) [Redux Thunk](https://github.com/reduxjs/redux-thunk)
8) [Redux/React Official Real World Example](https://github.com/reduxjs/redux/tree/master/examples/real-world)
