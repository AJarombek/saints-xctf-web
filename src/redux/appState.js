/**
 * Sample Redux State for the entire application.
 * @author Andrew Jarombek
 * @since 5/3/2020
 */

const appState = {
  auth: {
    auth: {
      isFetching: false,
      lastUpdated: 1588530309,
      signedIn: true,
      status: "SUCCESS"
    },
    registration: {
      stage: 1,
      first: "Andy",
      last: "Jarombek",
      email: "andrew@jarombek.com",
      username: undefined,
      password: undefined,
      activationCode: undefined,
      teams: {
        saintsXCTFAlumni: {
          status: "accepted",
          groups: {
            alumni: {
              status: "accepted"
            }
          }
        }
      }
    },
    user: {
      andy: {
        isFetching: false,
        didInvalidate: false,
        lastUpdated: 1588530310,
        username: "andy",
        first: "Andrew",
        last: "Jarombek",
        password: "xxxxxx",
        description: "Test Description",
        memberSince: "2016-02-23 12:00:00",
        classYear: 2017,
        location: "New York, NY",
        favoriteEvent: "Cooldown Jog",
        activationCode: "ABCD1234",
        email: "andrew@jarombek.com",
        lastSignIn: "2020-05-03 12:00:00",
        subscribed: "Y",
        logs: []
      }
    },
  },
  profile: {
    tom: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530311
    },
    joe: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530312
    },
    ben: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530313
    }
  },
  group: {
    wmenstf: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530314,
      groupName: "wmenstf",
      groupTitle: "Women's Track & Field",
      description: "",
      weekStart: "monday",
      members: [],
      logs: []
    }
  },
  logs: {
    isFetching: false,
    didInvalidate: false,
    lastUpdated: 1588530315,
    items: {
      1: {
        log_id: 1,
        username: "andy",
        first: "Andrew",
        last: "Jarombek",
        name: "Reservoir Loop",
        location: "New York, NY",
        date: "2020-05-03",
        type: "run",
        distance: 6.37,
        metric: "miles",
        miles: 6.37,
        time: "00:45:37",
        pace: "00:07:10",
        feel: 6,
        description: "",
        timeCreated: "2020-05-03 12:00:00",
        comments: {
          1: {
            commentId: 1,
            username: "andy",
            first: "Andrew",
            last: "Jarombek",
            logId: 1,
            time: "2020-05-03 12:00:00",
            content: "test"
          }
        }
      }
    }
  }
};
