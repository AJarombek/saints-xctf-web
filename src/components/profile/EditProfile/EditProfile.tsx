/**
 * Component for a tab on the profile page which allows users to edit their profile.
 * @author Andrew Jarombek
 * @since 10/18/2020
 */

import React, { useEffect, useRef, useState } from 'react';
import { createUseStyles } from 'react-jss';
import styles from './styles';
import { Memberships, RootState, UserMeta, Users } from '../../../redux/types';
import ImageInput, { ImageInputStatus } from '../../shared/ImageInput';
import classNames from 'classnames';
import AutoResizeTextArea from '../../shared/AutoResizeTextArea';
import RadioButton from '../../shared/RadioButton';
import UploadFile from '../../shared/UploadFile/UploadFile';
import { AJButton } from 'jarombek-react-components';
import { useDispatch, useSelector } from 'react-redux';
import { getUserMemberships } from '../../../redux/modules/profile';
import PickTeams from '../PickTeams';

interface Props {
  user: UserMeta;
}

const useStyles = createUseStyles(styles);

const EditProfile: React.FunctionComponent<Props> = ({ user }) => {
  const classes = useStyles();

  const dispatch = useDispatch();
  const userProfiles: Users = useSelector((state: RootState) => state.profile.users);

  const descriptionRef = useRef(null);

  const [memberships, setMemberships] = useState<Memberships>(null);
  const [firstName, setFirstName] = useState('');
  const [firstNameStatus, setFirstNameStatus] = useState<ImageInputStatus>(ImageInputStatus.NONE);
  const [lastName, setLastName] = useState('');
  const [lastNameStatus, setLastNameStatus] = useState<ImageInputStatus>(ImageInputStatus.NONE);
  const [email, setEmail] = useState('');
  const [emailStatus, setEmailStatus] = useState<ImageInputStatus>(ImageInputStatus.NONE);
  const [classYear, setClassYear] = useState(0);
  const [location, setLocation] = useState('');
  const [favoriteEvent, setFavoriteEvent] = useState('');
  const [description, setDescription] = useState('');
  const [weekStart, setWeekStart] = useState('');

  useEffect(() => {
    dispatch(getUserMemberships(user.username));
  }, [dispatch, user.username]);

  useEffect(() => {
    setMemberships(userProfiles[user.username].memberships);
  }, [userProfiles, user.username]);

  const onWeekStartChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    if (e.target.checked) {
      setWeekStart(e.target.value);
    }
  };

  const onSubmitDetails = (): void => {};

  const onCancelDetails = (): void => {
    setFirstName(user.first);
    setLastName(user.last);
    setEmail(user.email);
    setClassYear(user.class_year);
    setLocation(user.location);
    setFavoriteEvent(user.favorite_event);
    setDescription(user.description);
    setWeekStart(user.week_start);
  };

  const onSubmitPicture = (): void => {};
  const onCancelPicture = (): void => {};

  return (
    <div className={classes.editProfile}>
      <h3 className={classes.title}>Profile Details</h3>
      <div className={classes.form}>
        <div className={classes.twoInputs}>
          <div
            className={classNames(
              classes.firstNameInput,
              firstNameStatus === ImageInputStatus.FAILURE && classes.inputError
            )}
          >
            <p className={classes.inputTitle}>First Name*</p>
            <ImageInput
              type="text"
              name="first"
              placeholder=""
              useCustomValue={true}
              value={firstName}
              onChange={(e): void => setFirstName(e.target.value)}
              status={firstNameStatus}
            />
          </div>
          <div
            className={classNames(
              classes.lastNameInput,
              lastNameStatus === ImageInputStatus.FAILURE && classes.inputError
            )}
          >
            <p className={classes.inputTitle}>Last Name*</p>
            <ImageInput
              type="text"
              name="last"
              placeholder=""
              useCustomValue={true}
              value={lastName}
              onChange={(e): void => setLastName(e.target.value)}
              status={lastNameStatus}
            />
          </div>
        </div>
        <div className={classes.twoInputs}>
          <div
            className={classNames(classes.emailInput, emailStatus === ImageInputStatus.FAILURE && classes.inputError)}
          >
            <p className={classes.inputTitle}>Email*</p>
            <ImageInput
              type="email"
              name="email"
              placeholder=""
              useCustomValue={true}
              value={email}
              onChange={(e): void => setEmail(e.target.value)}
              status={emailStatus}
            />
          </div>
          <div className={classes.classYearInput}>
            <p className={classes.inputTitle}>Class Year</p>
            <ImageInput
              type="number"
              name="classYear"
              placeholder=""
              useCustomValue={true}
              value={`${classYear}`}
              onChange={(e): void => setClassYear(+e.target.value)}
              status={ImageInputStatus.NONE}
            />
          </div>
        </div>
        <div className={classes.twoInputs}>
          <div className={classes.locationInput}>
            <p className={classes.inputTitle}>Location</p>
            <ImageInput
              type="text"
              name="location"
              placeholder=""
              useCustomValue={true}
              value={location}
              onChange={(e): void => setLocation(e.target.value)}
              status={ImageInputStatus.NONE}
            />
          </div>
          <div className={classes.favoriteEventInput}>
            <p className={classes.inputTitle}>Favorite Event</p>
            <ImageInput
              type="number"
              name="favoriteEvent"
              placeholder=""
              useCustomValue={true}
              value={favoriteEvent}
              onChange={(e): void => setFavoriteEvent(e.target.value)}
              status={ImageInputStatus.NONE}
            />
          </div>
        </div>
        <div>
          <p className={classes.inputTitle}>Description</p>
          <AutoResizeTextArea
            maxLength={1000}
            placeholder="..."
            onChange={(e: React.ChangeEvent<HTMLTextAreaElement>): void => setDescription(e.target.value)}
            useCustomValue={true}
            value={description}
            disabled={false}
            className={classes.textArea}
            ref={descriptionRef}
          />
        </div>
        <div>
          <p className={classes.inputTitle}>Week Start</p>
          <div className={classes.radioGroup}>
            <RadioButton
              id="sunday"
              name="weekStart"
              value="sunday"
              label="Sunday"
              onChange={onWeekStartChange}
              defaultChecked={user.week_start === 'sunday'}
              className={classes.radio}
            />
            <RadioButton
              id="monday"
              name="weekStart"
              value="monday"
              label="Monday"
              onChange={onWeekStartChange}
              defaultChecked={user.week_start === 'monday'}
              className={classes.radio}
            />
          </div>
        </div>
        <div className={classes.actions}>
          <AJButton type="contained" disabled={false} onClick={onSubmitDetails} className={classes.submitButton}>
            Save Details
          </AJButton>
          <AJButton type="text" disabled={false} onClick={onCancelDetails} className={classes.cancelButton}>
            Cancel
          </AJButton>
        </div>
      </div>
      <h3 className={classes.title}>Profile Picture</h3>
      <div className={classes.form}>
        <div className={classes.profilePictureContainer}>
          <figure className={classes.picture}>
            <img src="" alt="" />
          </figure>
          <UploadFile />
        </div>
        <div className={classes.actions}>
          <AJButton type="contained" disabled={false} onClick={onSubmitPicture} className={classes.submitButton}>
            Save Picture
          </AJButton>
          <AJButton type="text" disabled={false} onClick={onCancelPicture} className={classes.cancelButton}>
            Cancel
          </AJButton>
        </div>
      </div>
      <h3 className={classes.title}>Teams and Groups</h3>
      <div className={classes.form}>
        <PickTeams teams={memberships?.teams} />
        <div className={classes.actions}>
          <AJButton type="contained" disabled={false} onClick={onSubmitPicture} className={classes.submitButton}>
            Save Teams & Groups
          </AJButton>
          <AJButton type="text" disabled={false} onClick={onCancelPicture} className={classes.cancelButton}>
            Cancel
          </AJButton>
        </div>
      </div>
    </div>
  );
};

export default EditProfile;
