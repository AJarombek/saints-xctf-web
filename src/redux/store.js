/**
 * Build and configure the redux store, which holds the application state and handles
 * all state updates.
 * @author Andrew Jarombek
 * @since 5/1/2020
 */

import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import createLogger from 'redux-logger';
import reducer from './modules/reducers';

const history = createBrowserHistory();

const configureStore = (data) => {
  const loggerMiddleware = createLogger();
  const middleware = [loggerMiddleware, routerMiddleware(history)];

  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(reducer, data);
};

export { history };
export default configureStore;
