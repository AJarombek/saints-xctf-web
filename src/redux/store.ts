/**
 * Build and configure the redux store, which holds the application state and handles
 * all state updates.
 * @author Andrew Jarombek
 * @since 5/1/2020
 */

import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware, Store, Action } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import { createLogger } from 'redux-logger';
import thunkMiddleware, { ThunkAction } from 'redux-thunk';
import reducer from './modules/reducers';
import { RootState } from './types';

export const history = createBrowserHistory();

export default function configureStore(): Store {
  const loggerMiddleware = createLogger();
  const middleware = [routerMiddleware(history), loggerMiddleware, thunkMiddleware];

  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  const store = createStoreWithMiddleware(reducer(history));

  if (process.env.NODE_ENV !== 'production' && module.hot) {
    module.hot.accept('./modules/reducers', () => store.replaceReducer(reducer));
  }

  return store;
}

export type AppThunk<ReturnType = void, State = RootState> = ThunkAction<ReturnType, State, unknown, Action<string>>;
