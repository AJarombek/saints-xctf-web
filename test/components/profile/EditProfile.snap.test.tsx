/**
 * Snapshot test for the EditProfile component.
 * @author Andrew Jarombek
 * @since 3/6/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Store } from 'redux';
import { emptyStore } from '../../test-utils/storeMocks';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { andy } from '../../test-utils/userMocks';
import EditProfile from '../../../src/components/profile/EditProfile';
import { MemoryRouter } from 'react-router-dom';

const mockStore = configureStore([thunk]);

describe('EditProfile Snapshot Tests', () => {
  let store: Store;

  beforeEach(() => {
    store = mockStore({
      ...emptyStore,
      profile: {
        users: {
          andy: {
            user: andy,
            uploadingProfilePicture: {
              uploaded: false,
              uploadedSize: null,
              totalSize: null
            }
          }
        }
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <EditProfile user={andy} />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
