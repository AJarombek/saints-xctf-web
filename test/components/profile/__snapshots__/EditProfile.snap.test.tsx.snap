// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile Snapshot Tests renders correctly 1`] = `
<div
  className="editProfile-0-1-1"
  id="editProfile"
>
  <h3
    className="title-0-1-2"
  >
    Profile Details
  </h3>
  <div
    className="form-0-1-3"
  >
    <div
      className="twoInputs-0-1-7"
    >
      <div
        className="firstNameInput-0-1-11"
      >
        <p
          className="inputTitle-0-1-6"
        >
          First Name*
        </p>
        <div
          className="sxctf-image-input"
          data-cypress="image-input-first"
        >
          <div
            className="no-icon"
          >
             
          </div>
          <input
            autoComplete=""
            maxLength={30}
            name="first"
            onChange={[Function]}
            placeholder=""
            type="text"
            value="Andy"
          />
          <div>
            <div
              className="status none"
            >
              <p />
            </div>
          </div>
        </div>
        <p
          className="inputTip-0-1-22"
          data-cypress="firstNameInputTip"
          hidden={true}
        >
          First name is a required field.
        </p>
      </div>
      <div
        className="lastNameInput-0-1-12"
      >
        <p
          className="inputTitle-0-1-6"
        >
          Last Name*
        </p>
        <div
          className="sxctf-image-input"
          data-cypress="image-input-last"
        >
          <div
            className="no-icon"
          >
             
          </div>
          <input
            autoComplete=""
            maxLength={30}
            name="last"
            onChange={[Function]}
            placeholder=""
            type="text"
            value="Jarombek"
          />
          <div>
            <div
              className="status none"
            >
              <p />
            </div>
          </div>
        </div>
        <p
          className="inputTip-0-1-22"
          data-cypress="lastNameInputTip"
          hidden={true}
        >
          Last name is a required field.
        </p>
      </div>
    </div>
    <div
      className="twoInputs-0-1-7"
    >
      <div
        className="emailInput-0-1-13"
      >
        <p
          className="inputTitle-0-1-6"
        >
          Email*
        </p>
        <div
          className="sxctf-image-input"
          data-cypress="image-input-email"
        >
          <div
            className="no-icon"
          >
             
          </div>
          <input
            autoComplete=""
            maxLength={50}
            name="email"
            onChange={[Function]}
            placeholder=""
            type="email"
            value="andrew@jarombek.com"
          />
          <div>
            <div
              className="status none"
            >
              <p />
            </div>
          </div>
        </div>
        <p
          className="inputTip-0-1-22"
          data-cypress="emailInputTip"
          hidden={true}
        >
          A valid email address is required.
        </p>
      </div>
      <div
        className="classYearInput-0-1-14"
      >
        <p
          className="inputTitle-0-1-6"
        >
          Class Year
        </p>
        <div
          className="sxctf-image-input"
          data-cypress="image-input-classYear"
        >
          <div
            className="no-icon"
          >
             
          </div>
          <input
            autoComplete=""
            maxLength={4}
            name="classYear"
            onChange={[Function]}
            placeholder=""
            type="number"
            value="2017"
          />
          <div>
            <div
              className="status none"
            >
              <p />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="twoInputs-0-1-7"
    >
      <div
        className="locationInput-0-1-15"
      >
        <p
          className="inputTitle-0-1-6"
        >
          Location
        </p>
        <div
          className="sxctf-image-input"
          data-cypress="image-input-location"
        >
          <div
            className="no-icon"
          >
             
          </div>
          <input
            autoComplete=""
            maxLength={50}
            name="location"
            onChange={[Function]}
            placeholder=""
            type="text"
            value="Riverside, CT"
          />
          <div>
            <div
              className="status none"
            >
              <p />
            </div>
          </div>
        </div>
      </div>
      <div
        className="favoriteEventInput-0-1-16"
      >
        <p
          className="inputTitle-0-1-6"
        >
          Favorite Event
        </p>
        <div
          className="sxctf-image-input"
          data-cypress="image-input-favoriteEvent"
        >
          <div
            className="no-icon"
          >
             
          </div>
          <input
            autoComplete=""
            maxLength={20}
            name="favoriteEvent"
            onChange={[Function]}
            placeholder=""
            type="text"
            value="5000m, 8K"
          />
          <div>
            <div
              className="status none"
            >
              <p />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <p
        className="inputTitle-0-1-6"
      >
        Description
      </p>
      <textarea
        className="textArea-0-1-23 textArea-0-1-8"
        disabled={false}
        maxLength={255}
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="..."
        value="I sometimes like to run."
      />
    </div>
    <div>
      <p
        className="inputTitle-0-1-6"
      >
        Week Start
      </p>
      <div
        className="radioGroup-0-1-9"
      >
        <div
          className="radio-0-1-24 radio-0-1-10"
          data-cypress="radioButton"
        >
          <label>
            <div
              className="inputWrapper-0-1-25"
            >
              <input
                className="input-0-1-26"
                defaultChecked={false}
                id="sunday"
                name="weekStart"
                onChange={[Function]}
                type="radio"
                value="sunday"
              />
              <div
                className="customRadio-0-1-27"
                data-cypress="customRadio"
              />
            </div>
            <div
              className="customLabel-0-1-28"
            >
              Sunday
            </div>
          </label>
        </div>
        <div
          className="radio-0-1-24 radio-0-1-10"
          data-cypress="radioButton"
        >
          <label>
            <div
              className="inputWrapper-0-1-25"
            >
              <input
                className="input-0-1-26"
                defaultChecked={true}
                id="monday"
                name="weekStart"
                onChange={[Function]}
                type="radio"
                value="monday"
              />
              <div
                className="customRadio-0-1-27"
                data-cypress="customRadio"
              />
            </div>
            <div
              className="customLabel-0-1-28"
            >
              Monday
            </div>
          </label>
        </div>
      </div>
    </div>
    <div
      className="actions-0-1-17"
    >
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonDisabled-2-2-6 aj-contained-button ajContainedButton-2-2-1 ajContainedButtonDisabled-2-2-3 submitButton-0-1-18 disabledSubmitButton-0-1-20"
        onClick={null}
      >
        <button
          disabled={true}
          type="button"
        >
          <p>
            Save Details
          </p>
        </button>
      </div>
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonEnabled-2-2-5 aj-text-button ajTextButton-2-2-7 ajTextButtonEnabled-2-2-8 cancelButton-0-1-19"
        onClick={[Function]}
      >
        <button
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
  <h3
    className="title-0-1-2"
  >
    Profile Picture
  </h3>
  <div
    className="form-0-1-3"
  >
    <div
      className="pictureContainer-0-1-29"
    >
      <figure
        className="picture-0-1-30"
      >
        <img
          alt=""
          src="/asset/saintsxctf.png"
        />
      </figure>
      <div
        className="uploadFile-0-1-37 uploadFileBoxEmpty-0-1-39"
        data-cypress="uploadFile"
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
      >
        <input
          accept="image/*"
          className="hiddenInput-0-1-40"
          onChange={[Function]}
          type="file"
        />
        <p
          className="uploadText-0-1-41 uploadTextEmpty-0-1-43"
        >
          Click here or drag and drop a file.
        </p>
      </div>
    </div>
    <div
      className="note-0-1-36"
    >
      <p>
        NOTE: For best results, crop your image into a square.
      </p>
    </div>
    <div
      className="actions-0-1-31"
    >
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonDisabled-2-2-6 aj-contained-button ajContainedButton-2-2-1 ajContainedButtonDisabled-2-2-3 submitButton-0-1-32 disabledSubmitButton-0-1-33"
        onClick={null}
      >
        <button
          disabled={true}
          type="button"
        >
          <p>
            Save Picture
          </p>
        </button>
      </div>
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonEnabled-2-2-5 aj-text-button ajTextButton-2-2-7 ajTextButtonEnabled-2-2-8 cancelButton-0-1-35"
        onClick={[Function]}
      >
        <button
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
  <h3
    className="title-0-1-2"
  >
    Teams and Groups
  </h3>
  <div
    className="form-0-1-3"
  >
    <div
      className="pickTeams-0-1-44"
    >
      <div
        className="search-0-1-45"
      >
        <div
          className="sxctf-image-input"
          data-cypress="image-input-team"
        >
          <div
            className="no-icon"
          >
             
          </div>
          <input
            autoComplete=""
            name="team"
            onChange={[Function]}
            placeholder="Search Teams (eg. SaintsXCTF Alumni)"
            type="text"
          />
          <div>
            <div
              className="status none"
            >
              <p />
            </div>
          </div>
        </div>
        <div
          className="searchedTeams-0-1-47"
        />
      </div>
      <div />
    </div>
    <div
      className="actions-0-1-49"
    >
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonDisabled-2-2-6 aj-contained-button ajContainedButton-2-2-1 ajContainedButtonDisabled-2-2-3 submitButton-0-1-50 disabledSubmitButton-0-1-51"
        onClick={null}
      >
        <button
          disabled={true}
          type="button"
        >
          <p>
            Save Teams & Groups
          </p>
        </button>
      </div>
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonEnabled-2-2-5 aj-text-button ajTextButton-2-2-7 ajTextButtonEnabled-2-2-8 cancelButton-0-1-53"
        onClick={[Function]}
      >
        <button
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;
